frontend/packages/core/src/AppProvider/workflow-hydrator.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
+import type { clutch as IClutch } from "@clutch-sh/api";
+import { Grid } from "@material-ui/core";
+
+import { WorkflowStorageContext } from "../Contexts/workflow-storage-context";
+import { retrieveData } from "../Contexts/workflow-storage-context/helpers";
+import workflowStorageContextReducer from "../Contexts/workflow-storage-context/reducer";
+import type { WorkflowStorageContextProps } from "../Contexts/workflow-storage-context/types";
+import { defaultWorkflowStorageState } from "../Contexts/workflow-storage-context/types";
+import { Alert } from "../Feedback";
+import styled from "../styled";
+
+interface WorkflowHydratorProps {
+  children: React.ReactElement;
+  hydrateData: IClutch.shortlink.v1.IShareableState[];
+  onClear: () => void;
```

Comment:
just from looking at the props it's unclear when or why this is called

Reply:
This is called so that we can clear the data temporarily stored in AppProvider after initialization. I've added docstrings around this as well as renamed it.